Engine/csound_orc_compile.c

```diff
@@ -1935,10 +1935,14 @@ if (engineState != &csound->engineState) {
       continue;
     if (PARSER_DEBUG)
       csound->DebugMsg(csound, "Instr 0 check on opcode=%s\n", bp->t.opcod);
-    if (UNLIKELY((thread = oentry->thread) & 06 ||
-                 (!thread && bp->t.pftype != 'b'))) {
-      csound->DebugMsg(csound, "***opcode=%s thread=%d pftype=%c\n",
-                       bp->t.opcod, thread, bp->t.pftype);
+    /* VL: now the check is simply for oentry->perf, which is the
+       only condition possible for perf-time code 
+    */
+    if (UNLIKELY((oentry->perf != NULL) ||
```

Comment:
This check still seems off to me, if perf != NULL, it'll never make it to the second conditional check.

Reply:
yep, it's much simpler.